utils/config.mlp

```diff
@@ -135,9 +138,10 @@ let print_config oc =
   p "standard_library" standard_library;
   p "standard_runtime" standard_runtime;
   p "ccomp_type" ccomp_type;
-  p "bytecomp_c_compiler" bytecomp_c_compiler;
+  p "c_compiler" c_compiler;
```

Comment:
This breaks all build scripts that parse the output of `ocamlc -config` in order to determine the c compiler - but there is no entry in `Changes` 😒

Reply:
Oops - although how many packages? One could of course argue it reveals how many packages incorrectly parse the output `ocamlc -config` to compile C stubs rather than correctly calling ocamlc/ocamlopt on a C file!

Reply:
Yes: for example, the ctypes build is [broken](https://travis-ci.org/ocamllabs/ocaml-ctypes/jobs/279517112#L3259) as a result of this change, since the ctypes makefile [uses the output of `ocamlc -config`](https://github.com/ocamllabs/ocaml-ctypes/blob/8e9fe43ecc4257cfc71a5f744f8708feeeb03b39/Makefile.tests#L7).

Reply:
Is it possible to change the relevant build scripts to use `ocamlc` as the C compiler, as @dra27? If this work, it is both sensible and backward-compatible, so that may be a nice fix. Otherwise, we may want to re-export `{bytecomp,native}_c_compiler` in -config (defined/derived from `c_compiler` and `ocaml{c,opt}_cflags`) for compatibility.

Reply:
I created a Mantis issue to discuss this problem. [MPR#7650](https://caml.inria.fr/mantis/view.php?id=7650). Let's move the discussion there instead of lost in a github commit comment.

  https://caml.inria.fr/mantis/view.php?id=7650